docs/guides/deployment/docker.md

```diff
@@ -12,129 +12,227 @@ search:
+import httpx
+from prefect import flow
 
-## Storing Prefect Flow Code in a Docker Image 
 
-First let's create a directory to work from, `docker-tutorial`.
+@flow(log_prints=True)
+def get_repo_info(repo_name: str = "PrefectHQ/prefect"):
+    url = f"https://api.github.com/repos/{repo_name}"
+    response = httpx.get(url)
+    response.raise_for_status()
+    repo = response.json()
+    print(f"{repo_name} repository statistics 🤓:")
+    print(f"Stars 🌠 : {repo['stargazers_count']}")
+    print(f"Forks 🍴 : {repo['forks_count']}")
 
-In this directory, you will create a sub-directory named `flows` and put your flow script from the [Deployments](/tutorial/deployments/) tutorial. In this case, I've named the flow `docker-tutorial-flow.py`.
 
-The next file you will add to the `docker-tutorial` directory is a `requirements.txt`.  In this file make sure to include all dependencies that are required for your `docker-tutorial-flow.py` script.  
+if __name__ == "__main__":
+    get_repo_info.serve(name="prefect-docker-guide")
+```
 
-Next, you will create a `Dockerfile` that will be used to create a Docker image that will also store the flow code.  This `Dockerfile` should look similar to the following:  
+The next file we'll add to the `prefect-docker-guide` directory is a `requirements.txt`. We'll include all dependencies required for our `prefect-docker-guide-flow.py` script in the Docker image we'll build.
 
+<div class="terminal">
 ```bash
+touch requirements.txt
+```
+</div>
+
+Here's what we'll put in our `requirements.txt` file:
+
+```txt title="requirements.txt"
+httpx
+```
+
+Next, we'll create a `Dockerfile` that we'll use to create a Docker image that will also store the flow code. 
+
+<div class="terminal">
+```bash
+touch Dockerfile
+```
+</div>
+
+We'll add the following content to our `Dockerfile`:
+
+```dockerfile title="Dockerfile"
+# We're using the latest version of Prefect with Python 3.9
 FROM prefecthq/prefect:2-python3.9
+
+# Add our requirements.txt file to the image and install dependencies
 COPY requirements.txt .
 RUN pip install -r requirements.txt --trusted-host pypi.python.org --no-cache-dir
-ADD flows /opt/prefect/flows
+
+# Add our flow code to the image
+COPY flows /opt/prefect/flows
+
+# Run our flow script when the container starts
+CMD ["python", "flows/prefect-docker-guide-flow.py"]
 ```
 
-Finally, we build this image by running: 
+## Building a Docker image
+
+Now that we have a Dockerfile we can build our image by running: 
 
+<div class="terminal">
 ```bash
-docker build -t docker-tutorial-image .
+docker build -t prefect-docker-guide-image .
 ```
+</div>
 
-This will create a Docker image in your Docker repository with your Prefect flow stored in the image.  
+And we can check that our image works by running:
 
-## Prefect.yaml File
+<div class="terminal">
+```bash
+docker run -e PREFECT_API_URL=YOUR_PREFECT_API_URL -e PREFECT_API_KEY=YOUR_API_KEY prefect-docker-guide-image
```

Comment:
I think you should make this section two tabs: one for Cloud (your command here) and one for self-hosted which requires a bit more work.  If you're self hosting you need to do some port-forwarding / networking to make sure the container is talking to your machine's API (I believe there's one or two additional flags that you need) - this should already be documented somewhere we just need to find it.